kinesis/kinesis.go

```diff
@@ -144,28 +144,31 @@ func NewOutputPlugin(region, stream, dataKeys, partitionKey, roleARN, endpoint,
 }
 
 // newPutRecordsClient creates the Kinesis client for calling the PutRecords method
-func newPutRecordsClient(roleARN string, awsRegion string, endpoint string) (*kinesis.Kinesis, error) {
+func newPutRecordsClient(roleARN string, awsRegion string, kinesisEndpoint string, stsEndpoint string) (*kinesis.Kinesis, error) {
+	defaultResolver := endpoints.DefaultResolver()
```

Comment:
Can this be moved into the `customResolverFn` function definition instead of referenced via closure.

Reply:
It can be; I did it this way because that's how the example I copied did it

Reply:
Wait. If I do it in the function, then each invocation creates a new defaultResolver object, whereas if I do it via closure there's only one. 

I wonder if that's why the example chooses to use closure?

Reply:
Hmm could be. The [docs](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#hdr-Using_Custom_Endpoints) show an inline `endpoints.DefaultResolver().EndpointFor(...)`. It'd depend on how often the resolver function is called which I assume would be tied to session creation, but I'd need to dig to be confident. It seems super minor either way so feel free to move forward.

Reply:
I see, I'll change it then.